plugin/externalPluginEntry_test.go

```diff
@@ -661,6 +661,109 @@ func (suite *ExternalPluginEntryTestSuite) TestListReadWithMethodResults() {
 	}
 }
 
+type mockCommand struct {
+	mock.Mock
+}
+
+func (m *mockCommand) Start() error               { return m.Called().Error(0) }
+func (m *mockCommand) Run() error                 { return m.Called().Error(0) }
+func (m *mockCommand) Terminate()                 { m.Called() }
+func (m *mockCommand) Wait() error                { return m.Called().Error(0) }
+func (m *mockCommand) SetStdout(stdout io.Writer) { m.Called(stdout) }
+func (m *mockCommand) SetStderr(stderr io.Writer) { m.Called(stderr) }
+func (m *mockCommand) SetStdin(stdin io.Reader)   { m.Called(stdin) }
+func (m *mockCommand) StdoutPipe() (io.ReadCloser, error) {
+	args := m.Called()
+	return args.Get(0).(io.ReadCloser), args.Error(1)
+}
+func (m *mockCommand) StderrPipe() (io.ReadCloser, error) {
+	args := m.Called()
+	return args.Get(0).(io.ReadCloser), args.Error(1)
+}
+func (m *mockCommand) ExitCode() int { return m.Called().Int(0) }
+
+var _ = internal.Command(&mockCommand{})
+
+func (suite *ExternalPluginEntryTestSuite) TestWrite() {
```

Comment:
Why does Write have so many tests? Seems like it should be symmetric with other `InvokeAndWait` tests.

Reply:
I'll have to spend a little time on this tomorrow. I'm not sure the other tests are verifying all the assertions we expect. They also mock at a higher level (the script's InvokeAndWait call) than Write had available originally.